plugins/woocommerce-docs/src/Blocks/BlockConverter.php

```diff
@@ -0,0 +1,161 @@
+use League\CommonMark\MarkdownConverter;
+
+/**
+ * Class MarkdownParser
+ */
+class BlockConverter {
+
+	/** // phpcs:ignore Generic.Commenting.DocComment.MissingShort
+	 *
+	 * @var MarkdownParser The MarkdownParser instance.
+	 */
+	private $parser;
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$environment = new Environment();
+		$environment->addExtension( new CommonMarkCoreExtension() );
+		$this->parser = new MarkdownConverter( $environment );
+	}
+
+
+	/**
+	 * Convert Markdown to Gutenberg blocks.
+	 *
+	 * @param string $content The Markdown content.
+	 *
+	 * @return string
+	 */
+	public function convert( $content ) {
+		$html = $this->parser->convert( $content )->__toString();
+		return $this->convert_html_to_blocks( $html );
+	}
+
+	/**
+	 * Convert HTML to blocks.
+	 *
+	 * @param string $html The HTML content.
+	 */
+	private function convert_html_to_blocks( $html ) {
+		$blocks_html = '';
+		$dom         = new \DOMDocument();
+
+		$dom->loadHTML( $html );
+		$xpath = new \DOMXPath( $dom );
+		$nodes = $xpath->query( '//body/*' );
+
+		foreach ( $nodes as $node ) {
+			$blocks_html .= $this->convert_node_to_block( $node );
+		}
+
+		return $blocks_html;
+	}
+
+	/**
+	 * Convert a DOM node to a block.
+	 *
+	 * @param \DOMNode $node The DOM node.
+	 */
+	private function convert_node_to_block( $node ) {
+		$node_name    = $node->nodeName; // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+		$node_value   = $node->nodeValue; // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+		$node_content = $this->convert_child_nodes_to_blocks( $node );
+
+		switch ( $node_name ) {
+			case 'p':
+				return $this->create_block( 'paragraph', $node_name, $node_content );
+			case 'h1':
+				return $this->create_block( 'heading', $node_name, $node_content, array( 'level' => 1 ) );
+			case 'h2':
+				return $this->create_block( 'heading', $node_name, $node_content, array( 'level' => 2 ) );
+			case 'h3':
+				return $this->create_block( 'heading', $node_name, $node_content, array( 'level' => 3 ) );
+			case 'h4':
+				return $this->create_block( 'heading', $node_name, $node_content, array( 'level' => 4 ) );
+			case 'h5':
+				return $this->create_block( 'heading', $node_name, $node_content, array( 'level' => 5 ) );
+			case 'h6':
+				return $this->create_block( 'heading', $node_name, $node_content, array( 'level' => 6 ) );
+			case 'ul':
+				return $this->create_block( 'list', $node_name, $node_content, array( 'ordered' => false ) );
+			case 'ol':
+				return $this->create_block( 'list', $node_name, $node_content, array( 'ordered' => true ) );
+			case 'li':
+				return $this->create_block( 'list-item', $node_name, $node_content );
+			case 'hr':
+				return $this->create_block( 'separator', $node_name, null );
+			default:
+				return $this->create_block( 'paragraph', $node_value );
+		}
```

Comment:
Is it worthwhile to add support for a table, such as this one?

https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/dependency-extraction-webpack-plugin/README.md

Reply:
@psealock we can introduce other blocks as follow ups. On first glance it looks like table would be straightforward, but there is no rush I think to bloat the PR any more. I tried to keep it to the simplest minimal set of block support for a first cut.